templates/master/00-master/openstack/files/openstack-mdns-publisher.yaml

```diff
@@ -0,0 +1,77 @@
+filesystem: "root"
+mode: 0644
+path: "/etc/kubernetes/manifests/mdns-publisher.yaml"
+contents:
+  inline: |
+    kind: Pod
+    apiVersion: v1
+    metadata:
+      name: mdns-publisher
+      namespace: kube-system
+      creationTimestamp:
+      deletionGracePeriodSeconds: 65
+      labels:
+        app: kni-infra-mdns
+    spec:
+      volumes:
+      - name: resource-dir
+        hostPath:
+          path: "/etc/kubernetes/static-pod-resources/mdns"
+      - name: conf-dir
+        hostPath:
+          path: "/etc/kubernetes/static-pod-resources/mdns"
+      initContainers:
+      - name: render-config
+        image: quay.io/openshift/origin-node:latest
+        command:
+        - "/bin/bash"
+        - "-c"
+        - |
+          #/bin/bash
+          set -ex
+
+          CLUSTER_NAME={{(split "." .EtcdDiscoveryDomain)._0}}
+          DOMAIN={{.EtcdDiscoveryDomain}}
+          API_VIP="$(dig +noall +answer "api.${DOMAIN}" | awk '{print $NF}')"
+          SUBNET_CIDR="$(ip addr show | grep -v "scope host" | grep -Po 'inet \K[\d.]+/[\d.]+' | head -n1)"
+          PREFIX="${SUBNET_CIDR#*/}"
+          NON_VIRTUAL_IP="${SUBNET_CIDR%/*}"
+          MASTER_HOSTNAME="$(hostname -s | cut -d '-' -f 4)-$(hostname -s | cut -d '-' -f 5).local."
+          ETCD_HOSTNAME="$(echo "$MASTER_HOSTNAME" | sed 's;master;etcd;')"
+          export MASTER_HOSTNAME
+          export ETCD_HOSTNAME
+          export NON_VIRTUAL_IP
+          export DOMAIN
+          export CLUSTER_NAME
+          /usr/libexec/platform-python -c "from __future__ import print_function
```

Comment:
I think the more correct approach is for the image to install the Python it wants.